packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -216,9 +287,40 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     final Completer<XFile> _completer = Completer<XFile>();
     // Observe the input until we can return something
     input.onChange.first.then((event) {
-      final objectUrl = _handleOnChangeEvent(event);
-      if (!_completer.isCompleted && objectUrl != null) {
-        _completer.complete(XFile(objectUrl));
+      final file = _handleOnChangeEvent(event);
+      if (!_completer.isCompleted && file != null) {
+        _completer.complete(XFile(html.Url.createObjectUrl(file),
+            name: file.name,
+            length: file.size,
+            mimeType: file.type,
+            lastModified: file.lastModifiedDate));
+      }
+    });
+    input.onError.first.then((event) {
+      if (!_completer.isCompleted) {
+        _completer.completeError(event);
+      }
+    });
+    // Note that we don't bother detaching from these streams, since the
+    // "input" gets re-created in the DOM every time the user needs to
+    // pick a file.
+    return _completer.future;
+  }
+
+  Future<List<XFile>> _getSelectedXFiles(html.FileUploadInputElement input) {
```

Comment:
There's a lot of logic duplication in the code:

* `_getSelectedFile` and `_getSelectedFiles`, `_handleOnChangeEvent` and `_handleOnChangeEventForMultipleFiles`, etc...

I think the best way to unify logic is to **assume that the user is always selecting multiple files**, and then in the `getImage` just return `.first` of the array of XFiles.

That way we can reduce the size of this change by about half. (Tests would remain the same).

What do you think?

Reply:
i have removed the _getSelectedFiles method and i m now returning List<html.File>? from _handleOnChangeEvent.
_getSelectedFile  will now use the first file from _handleOnChangeEvent.
getFile is now returning the first file from _getSelectedXFiles.